tools/src/main/scala/scala/scalanative/nir/parser/NirParser.scala

```diff
@@ -0,0 +1,451 @@
+  }
+
+  object Local extends NirParser[nir.Local] {
+    override def parser: P[nir.Local] =
+      P("%" ~ id ~ "." ~ int map {
+        case (scope, id) => nir.Local(scope, id)
+      })
+  }
+
+  object Next extends NirParser[nir.Next] {
+    def Label =
+      P(Local.parser ~ ("(" ~ Val.parser.rep(sep = ",") ~ ")").? map {
+        case (name, args) => nir.Next.Label(name, args getOrElse Seq())
+      })
+    def Succ = P("succ" ~ Local.parser map (nir.Next.Succ(_)))
+    def Fail = P("fail" ~ Local.parser map (nir.Next.Fail(_)))
+    def Case =
+      P("case" ~ Val.parser ~ "=>" ~ Local.parser map {
+        case (value, name) => nir.Next.Case(value, name)
+      })
+    override def parser: P[nir.Next] = Label | Succ | Fail | Case
+  }
+
+  object Op extends NirParser[nir.Op] {
+    def Call =
+      P(
+          "call[" ~ Type.parser ~ "]" ~ Val.parser ~ "(" ~ Val.parser.rep(
+              sep = ",") ~ ")" map {
+        case (ty, f, args) => nir.Op.Call(ty, f, args)
+      })
+    def Load =
+      P("load[" ~ Type.parser ~ "]" ~ Val.parser map {
+        case (ty, ptr) => nir.Op.Load(ty, ptr)
+      })
+    def Store =
+      P("store[" ~ Type.parser ~ "]" ~ Val.parser ~ "," ~ Val.parser map {
+        case (ty, ptr, value) => nir.Op.Store(ty, ptr, value)
+      })
+    def Elem =
+      P(
+          "elem[" ~ Type.parser ~ "]" ~ Val.parser ~ "," ~ Val.parser.rep(
+              sep = ",") map {
+        case (ty, ptr, indices) => nir.Op.Elem(ty, ptr, indices)
+      })
+    def Extract =
+      P("extract" ~ Val.parser ~ "," ~ int.rep(sep = ",") map {
+        case (aggr, indices) => nir.Op.Extract(aggr, indices)
+      })
+    def Insert =
+      P(
+          "insert" ~ Val.parser ~ "," ~ Val.parser ~ "," ~ int
+            .rep(sep = ",") map {
+        case (aggr, value, indices) => nir.Op.Insert(aggr, value, indices)
+      })
+    def Stackalloc =
+      P("stackalloc[" ~ Type.parser ~ "]" ~ Val.parser.? map {
+        case (ty, n) => nir.Op.Stackalloc(ty, n getOrElse nir.Val.None)
+      })
+    def Bin =
+      P(NirParser.Bin.parser ~ "[" ~ Type.parser ~ "]" ~ Val.parser ~ "," ~ Val.parser map {
+        case (name, ty, l, r) => nir.Op.Bin(name, ty, l, r)
+      })
+    def Comp =
+      P(NirParser.Comp.parser ~ "[" ~ Type.parser ~ "]" ~ Val.parser ~ "," ~ Val.parser map {
+        case (name, ty, l, r) => nir.Op.Comp(name, ty, l, r)
+      })
+    def Conv =
+      P(NirParser.Conv.parser ~ "[" ~ Type.parser ~ "]" ~ Val.parser map {
+        case (name, ty, v) => nir.Op.Conv(name, ty, v)
+      })
+    def Select =
+      P("select" ~ Val.parser ~ "," ~ Val.parser ~ "," ~ Val.parser map {
+        case (cond, thenp, elsep) => nir.Op.Select(cond, thenp, elsep)
+      })
+    def Classalloc = P("classalloc" ~ Global.parser map (nir.Op.Classalloc(_)))
+    def Field =
+      P("field[" ~ Type.parser ~ "]" ~ Val.parser ~ "," ~ Global.parser map {
+        case (ty, value, name) => nir.Op.Field(ty, value, name)
+      })
```

Comment:
We actually never use the type in `Op.Field`. Lets remove it as a field altogether.